app/models/host.rb

```diff
@@ -22,7 +22,43 @@ class Host < ApplicationRecord
   validates :name, presence: true
   validates :account, presence: true
 
+  def in_inventory
+    ::HostInventoryAPI.new(
+      User.current.account,
+      ::Settings.host_inventory_url,
+      nil
+    ).inventory_host(id)
+  end
+
   class << self
+    def host_in_inventory(host_id)
+      new(id: host_id).in_inventory
+    end
+
+    def find_or_create_hosts_by_inventory_ids(ids)
+      ids.map { |id| find_or_create_from_inventory(id) }
+    end
+
+    def find_or_create_from_inventory(host_id)
+      existing_hosts = ::Pundit.policy_scope(User.current, self)
+                               .where(id: [host_id])
+
+      existing_hosts.first || create_from_inventory(host_id)
```

Comment:
`where(...).first` and then checking for `nil` can just use `find_by(...)` instead